### PR TITLE
feat: Add Home Depot store locator scraper (GraphQL API)

### PR DESCRIPTION
**Pull request by vmud**
## Summary

- Add Home Depot scraper using GraphQL Federation Gateway API with two-phase approach: discover ~2,021 stores by state (54 API calls), then enrich each with coordinates, services, hours, and flags
- Uses `ScrapeRunner` orchestration pattern with `RichURLCache` for Phase 1 caching and county merging between phases
- 38 unit tests covering dataclass, GraphQL retry logic, discovery, extraction, null safety, and `run()` entry point

## New Files

- `config/homedepot_config.py` — GraphQL queries, headers, state codes, constants
- `src/scrapers/homedepot.py` — `HomeDepotStore` dataclass (44 fields), `_post_graphql()` with retry, `discover_stores()`, `extract_store_details()`, `run()`
- `tests/test_scrapers/test_homedepot.py` — 38 tests across 6 test classes

## Modified Files

- `src/scrapers/__init__.py` — Added `homedepot` to `SCRAPER_REGISTRY`
- `config/retailers.yaml` — Added homedepot retailer config + concurrency entry
- `CLAUDE.md` — Added homedepot to architecture section

## Key Design Decisions

- **ScrapeRunner pattern** (matches AT&T refactor) instead of manual `scraper_utils`
- **Per-scraper dataclass** with 44 fields (11 services, 3 flags, 4 phone types) — too retailer-specific for generic `Store` schema
- **Hybrid hours**: 7 main columns as `"HH:MM-HH:MM"` strings + 3 JSON specialty hours columns
- **Local `_post_graphql()`**: No shared POST-with-retry exists; mirrors `get_with_retry()` semantics with GraphQL error checking
- **County merging**: Only Phase 1 returns county; passed through `ScrapeRunner` item dict to Phase 2

## Code Review Fixes (commit 5/5)

All 8 findings from parallel code review agents addressed:
- CRITICAL: Config import shadow → renamed to `hd_config`
- HIGH: Unmocked `time.sleep` in tests → 27% test speedup
- HIGH: YAML delays not forwarded → added `select_delays()`
- MEDIUM: Unredacted exception → `redact_credentials()`
- MEDIUM: Missing network exception test → added
- MEDIUM: Null `storeSearch` safety → `or {}` + test
- LOW: URL validation → `startswith("/l/")` check
- LOW: Dynamic `[{retailer}]` log prefix

## Test plan

- [x] 38 unit tests passing (pytest)
- [x] All pre-commit hooks passing (bandit, detect-secrets, safety)
- [ ] `python run.py --retailer homedepot --test` (10-store smoke test)
- [ ] `python run.py --retailer homedepot --limit 50` (limited run)
- [ ] Validate output JSON structure and field completeness

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Medium Risk**
> Adds a new network-facing scraper with custom retry/backoff and parallel extraction, which could impact run time/load or fail under API/anti-bot changes, but is largely isolated behind new files and covered by extensive unit tests.
> 
> **Overview**
> Adds **Home Depot** as a new supported retailer, including YAML configuration (concurrency, delays, output schema) and registry wiring so it can run via the standard CLI/runner flow.
> 
> Introduces a two-phase GraphQL scraper: state-by-state discovery (`storeDirectoryByState`) followed by per-store enrichment (`storeSearch`) with retry/backoff handling, null-safety, county carry-over between phases, and export-friendly formatting (stringified coords + JSON-encoded specialty hours). Includes a large unit test suite covering retry behavior, parsing, edge cases, and `run()` integration with `ScrapeRunner`/`RichURLCache`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 10b0eb2f62cd4a7b328bf4ba4587a753d4b6358f. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

## Summary by CodeRabbit

* **New Features**
  * Added Home Depot scraping: discovery + detailed GraphQL-based extraction of stores, hours, services, contact, and metadata.

* **Tests**
  * Added comprehensive tests covering data parsing, formatting, GraphQL behavior, retries, error handling, and end-to-end workflow.

* **Chores**
  * Updated scraper registry and retailer configuration (concurrency, delays, output fields) to enable Home Depot.

* **Documentation**
  * Project overview and architecture docs updated to include Home Depot.

### CODE REVIEW

File: CLAUDE.md [hunk 1]
```diff
@@ -134,7 +134,8 @@ run.py                          # Main CLI entry point - handles arg parsing, co
 │   ├── bestbuy.py              # XML sitemap
 │   ├── telus.py                # Uberall API (Canadian)
 │   ├── cricket.py              # Yext API (US)
-│   └── bell.py                 # Sitemap + JSON-LD (Canadian)
+│   ├── bell.py                 # Sitemap + JSON-LD (Canadian)
+│   └── homedepot.py            # GraphQL Federation Gateway API
```
[the remark this answers is not in the document]
Comment (vmud) [reply CLAUDE.md]: Fixed in commit 10b0eb2. Added Home Depot to Project Overview retailer list and test_homedepot.py to the Test Structure listing.
